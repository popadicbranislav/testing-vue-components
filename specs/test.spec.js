import TestComponent from "@/test";
import List from "@/list";
import { mount } from "@vue/test-utils";

test("mount a Test component", () => {
  const wrapper = mount(TestComponent, {
    propsData: {
      value: "VueSchool",
    },
  });
  expect(wrapper).toMatchSnapshot();
});

test("list component", async () => {
  const wrapper = mount(List);
  const movies = wrapper.vm.marvelMovies;
  await wrapper.setData({ marvelMovies: [...movies, "Endgame"] });
  // await wrapper.setData({ marvelMovies: [] });

  expect(wrapper).toMatchSnapshot();
});
