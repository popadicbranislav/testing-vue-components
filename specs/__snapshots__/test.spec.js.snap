// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list component 1`] = `
<div>
  <ul>
    <li><strong>Iron Man</strong></li>
    <li><strong>Avengers</strong></li>
    <li><strong>Infitinity War</strong></li>
    <li><strong>Endgame</strong></li>
  </ul>
</div>
`;

exports[`mount a Test component 1`] = `
<div>
  <h1>Hello VueSchool</h1>
</div>
`;
